packages/foam-vscode/src/features/wikilink-reference-generation.ts

```diff
@@ -97,13 +97,24 @@ async function updateReferenceList(foam: NoteGraph) {
   }
 }
 
+enum LinkReferenceDefinitionsSetting {
+  withExtensions = "withExtensions",
+  withoutExtensions = "withoutExtensions",
+}
+
 async function generateReferenceList(foam: NoteGraph, doc: TextDocument): Promise<string[]> {
   const filePath = doc.fileName;
 
   const id = dropExtension(basename(filePath));
 
+  const linkDefinitionSetting: LinkReferenceDefinitionsSetting = workspace
+    .getConfiguration("foam.edit")
```

Comment:
What is `foam.edit` is not defined? Will this return null and crash, or return an empty config object?

Reply:
good point, I am not sure how it works in vscode. it should never be undefined, as we are defining it in the `package.json`, but will look into how configuration works in vscode a bit more closely